opm/parser/eclipse/EclipseState/Tables/JFuncTable.hpp

```diff
@@ -0,0 +1,115 @@
+/*
+ Copyright 2016  Statoil ASA.
+
+ This file is part of the Open Porous Media project (OPM).
+
+ OPM is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OPM is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_PARSER_ECLIPSE_ECLIPSESTATE_TABLES_JFUNCTABLE_HPP_
+#define OPM_PARSER_ECLIPSE_ECLIPSESTATE_TABLES_JFUNCTABLE_HPP_
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+
+#include <boost/multi_array.hpp>
+
+#include <iostream>
+
+namespace Opm {
+
+class DeckKeyword;
+
+class JFuncTable
+{
+public:
+
+    enum Flag {
```

Comment:
We should start using the "new" `enum class` consistently: http://www.cprogramming.com/c++11/c++11-nullptr-strongly-typed-enum-class.html